File: market/test/java/ru/yandex/market/billing/api/config/FunctionalTestConfig.java
package ru.yandex.market.billing.api.config;

import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.boot.test.mock.mockito.MockBean;
import org.springframework.context.annotation.Bean;

import ru.yandex.common.util.date.TestableClock;
import ru.yandex.market.billing.core.factoring.ContractPayoutFrequencyDao;
import ru.yandex.market.billing.security.config.PassportConfig;

/**
 * Основной конфиг для тестов
 */
@TestConfiguration
public class FunctionalTestConfig {
    @MockBean
    private PassportConfig passportConfig;
    @MockBean
    private MvcConfig mvcConfig;

    @MockBean
    private ContractPayoutFrequencyDao contractPayoutFrequencyDaoMock;

    @Bean
    public TestableClock clock() {
        return new TestableClock();
    }
}

File: market/test/kotlin/ru/yandex/market/mbi/orderservice/api/config/FunctionalTestConfig.kt
package ru.yandex.market.mbi.orderservice.api.config

import org.mockito.kotlin.mock
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Import
import org.springframework.context.annotation.Primary
import org.springframework.context.support.PropertySourcesPlaceholderConfigurer
import ru.yandex.market.logistics.logistics4shops.api.InternalOrderApi
import ru.yandex.market.mbi.orderservice.api.persistence.dao.yt.summary.WeeklyCompensatedOrdersDao
import ru.yandex.market.mbi.orderservice.common.config.CommonFunctionalTestConfig
import ru.yandex.market.mbi.orderservice.common.persistence.repository.pg.KeyValueRepository
import ru.yandex.market.mbi.orderservice.common.service.external.Logistics4ShopsApiService
import ru.yandex.market.mbi.orderservice.common.service.pg.EnvironmentService
import ru.yandex.market.mbi.orderservice.common.service.pg.PgEnvironmentService
import java.time.Duration

@Configuration
@Import(CommonFunctionalTestConfig::class)
open class FunctionalTestConfig {

    companion object {
        @JvmStatic
        @Bean
        fun propertyConfigurer(): PropertySourcesPlaceholderConfigurer =
            PropertySourcesPlaceholderConfigurer().apply {
                order = -1
                setIgnoreUnresolvablePlaceholders(false)
            }
    }

    @Bean
    open fun environmentService(keyValueRepository: KeyValueRepository): EnvironmentService {
        return PgEnvironmentService(
            keyValueRepository,
            durationAdjuster = { Duration.ofNanos(1) }
        )
    }

    @Bean
    open fun logistics4ShopsApiService(): Logistics4ShopsApiService {
        return mock {}
    }

    @Bean
    open fun internalOrderApi(): InternalOrderApi {
        return mock {}
    }

    @Primary
    @Bean
    open fun mockWeeklyCompensatedOrdersDao(): WeeklyCompensatedOrdersDao = mock { }
}
